src/Data/Text/Lazy/IO.hs

```diff
@@ -129,11 +130,11 @@ hGetLine = hGetLineWith L.fromChunks
 
 -- | Write a string to a handle.
 hPutStr :: Handle -> Text -> IO ()
-hPutStr h = mapM_ (T.hPutStr h) . L.toChunks
+hPutStr h = T.hPutStr h . stream
 
 -- | Write a string to a handle, followed by a newline.
 hPutStrLn :: Handle -> Text -> IO ()
-hPutStrLn h t = hPutStr h t >> hPutChar h '\n'
+hPutStrLn h = hPutStr h . streamLn
```

Comment:
Sorry, I have limited brainpower at the moment. Could we go for a more conservative alternative is to leave `Lazy.hPutStr` unchanged and implement `Lazy.hPutStrLn` by `Strict.hPutStr` for all chunks except the last, which would be printed using `Strict.hPutStrLn`?

Reply:
Done.